The patches allow for MODEL, DMD (compiler for the library) and
DDOC (documentation generator) to be overriden on the command line.
The patch to gzio.c adds the missing errno.h include, to make the code portable:
http://d.puremagic.com/issues/show_bug.cgi?id=5463
--- dmd2/src/dmd/posix.mak
+++ dmd2/src/dmd/posix.mak
@@ -29,7 +29,7 @@
 TK=tk
 ROOT=root
 
-MODEL=32
+MODEL?=32
 
 ifeq (OSX,$(TARGET))
     ## See: http://developer.apple.com/documentation/developertools/conceptual/cross_development/Using/chapter_3_section_2.html#//apple_ref/doc/uid/20002000-1114311-BABGCAAB
--- dmd2/src/druntime/posix.mak
+++ dmd2/src/druntime/posix.mak
@@ -20,12 +20,12 @@
     endif
 endif
 
-DMD=dmd
+DMD?=dmd
 
 DOCDIR=doc
 IMPDIR=import
 
-MODEL=32
+MODEL?=32
 
 DFLAGS=-m$(MODEL) -O -release -inline -nofloat -w -d -Isrc -Iimport -property
 UDFLAGS=-m$(MODEL) -O -release -nofloat -w -d -Isrc -Iimport -property
--- dmd2/src/phobos/etc/c/zlib/gzio.c
+++ dmd2/src/phobos/etc/c/zlib/gzio.c
@@ -8,6 +8,7 @@
 /* @(#) $Id$ */
 
 #include <stdio.h>
+#include <errno.h>
 
 #include "zutil.h"
 
--- dmd2/src/phobos/posix.mak
+++ dmd2/src/phobos/posix.mak
@@ -61,8 +61,6 @@
 
 # Variable defined in an OS-dependent manner (see below)
 CC =
-DMD =
-DDOC =
 CFLAGS =
 DFLAGS =
 
@@ -83,14 +81,14 @@
 # Set CC and DMD
 ifeq ($(OS),win32wine)
 	CC = wine dmc.exe
-	DMD = wine dmd.exe
+	DMD ?= wine dmd.exe
 	RUN = wine
 else
 	ifeq ($(OS),win32remote)
-		DMD = ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
+		DMD ?= ssh 206.125.170.138 "cd code/dmd/phobos && dmd"
 		CC = ssh 206.125.170.138 "cd code/dmd/phobos && dmc"
 	else
-		DMD = dmd
+		DMD ?= dmd
 		ifeq ($(OS),win32)
 			CC = dmc
 		else
@@ -141,7 +139,7 @@
 endif
 
 # Set DDOC, the documentation generator
-DDOC=$(DMD)
+DDOC?=$(DMD)
 
 # Set LIB, the ultimate target
 ifeq (,$(findstring win,$(OS)))
